src/dictionary/methods/accessors.hpp

```diff
@@ -0,0 +1,59 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+#include "realm/dictionary.hpp"
+#include "dictionary/collection/collection.hpp"
+#include "common/object/interfaces.hpp"
+
+namespace realm {
+namespace js {
+
+template <typename T>
+struct DictionaryGetterSetter{
+    IOCollection *collection;
+    using Value = js::Value<T>;
+    using JSMixedAPI = TypeMixed<T>;
+
+    void set(accessor::Arguments args){
+        auto context = args.context;
+        auto key = args.property_name.c_str();
+        try{
```

Comment:
Is this needed? I mean, won't the `wrap` of `RealmObjectClass<T>::set_property()` catch it (https://github.com/realm/realm-js/blob/master/src/js_realm_object.hpp#L89)?

Reply:
Yes but it will make the module untestable.

Leaving us two choices tested code or perpetuating a bad practice with the use of global variables.